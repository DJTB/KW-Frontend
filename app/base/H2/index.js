import styled from 'styled-components';
import { beta, headingRhythm } from 'shared/styles/typography';

const H2 = styled.h2`
  ${beta}
  ${headingRhythm}
`;

export default H2;
