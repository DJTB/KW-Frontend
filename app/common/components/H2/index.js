import styled from 'styled-components';
import { beta, headingRhythm } from 'common/styles/typography';

const H2 = styled.h2`
  ${beta}
  ${headingRhythm}
`;

export default H2;
