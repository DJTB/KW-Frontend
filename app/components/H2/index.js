import styled from 'styled-components';
import { fluidType } from 'shared/styles/utils';
import { minFontSize, mobileMod, desktopMod } from 'shared/styles/sizing';

// Modular scale
const mobileSize = +((mobileMod ** 2.5) * minFontSize).toPrecision(3);
const desktopSize = +((desktopMod ** 2.5) * minFontSize).toPrecision(3);

const H2 = styled.h2`
  ${fluidType(mobileSize, desktopSize)}
`;

export default H2;
