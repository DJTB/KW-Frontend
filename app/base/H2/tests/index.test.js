import 'jest-styled-components';
import { shallow } from 'enzyme';
import React from 'react';

import H2 from '../index';

describe('<H2 />', () => {
  it('should render a prop', () => {
    const id = 'testId';
    const renderedComponent = shallow(<H2 id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should match styling snapshot', () => {
    expect(shallow(<H2>text</H2>)).toMatchStyledComponentsSnapshot();
  });
});
