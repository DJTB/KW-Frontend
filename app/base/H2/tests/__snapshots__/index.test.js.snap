// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> should match styling snapshot 1`] = `
.TYBEs {
  font-size: 19.9px;
  -webkit-letter-spacing: -0.03em;
  -moz-letter-spacing: -0.03em;
  -ms-letter-spacing: -0.03em;
  letter-spacing: -0.03em;
  padding: 0.28125rem;
  margin-top: 0;
  margin-bottom: 0;
  line-height: 1.2;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

.TYBEs:not(:first-of-type) {
  margin-top: 0.3999em;
  margin-bottom: 0;
}

@media (min-width: 600px) and (max-width: 1800px) {
  .TYBEs {
    font-size: calc(19.9px + (33.2 - 19.9) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width: 1800px) {
  .TYBEs {
    font-size: 33.2px;
  }
}

@media (min-width: 600px) {
  .TYBEs {
    padding: 0.33325rem;
  }
}

<h2
  className="sc-bdVaJa TYBEs"
>
  text
</h2>
`;
